packages/js/src/ui/components/Renderer.tsx

```diff
@@ -67,33 +68,40 @@ type RendererProps = {
   preferencesFilter?: PreferencesFilter;
   routerPush?: RouterPush;
   novu?: Novu;
+  containerElement?: Node | null | undefined;
 };
 
 export const Renderer = (props: RendererProps) => {
   const nodes = () => [...props.nodes.keys()];
 
   onMount(() => {
-    const id = 'novu-default-css';
-    const el = document.getElementById(id);
+    const id = NOVU_DEFAULT_CSS_ID;
+    const root = props.containerElement instanceof ShadowRoot ? props.containerElement : document;
```

Comment:
if the `containerElement` is a shadow root, then use it to find the default styles